### PR TITLE
Boxer table

### PR DESCRIPTION
**Pull request by stevetait257**
Added feature to print out list of boxers on the index page

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -7,7 +7,7 @@
   <meta http-equiv="X-UA-Compatible" content="ie=edge" />
   <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css" />
   <link rel="stylesheet" href="css/styles.css" />
-  <script src="app.js" defer></script>
+  <script src="index.js" defer></script>
   <title>NO css</title>
```
Comment (fsodano) [inline index.html]: ?

File: index.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// alert('hey how ya going, mate?!');
```
Comment (fsodano) [inline index.js]: no comments

File: index.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+const boxerStorage = JSON.parse(localStorage.getItem('boxers'));
+const table = document.querySelector('#boxer-list');
+
+
+(function printBoxers() {
+  // debugger;
```
Comment (fsodano) [inline index.js]: nein!

File: index.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    ];
+
+    boxerData.forEach(el => {
+      const td = document.createElement('td');
+      td.textContent = [...el];
+      tr.appendChild(td);
+    });
+
+  });
+})();
```
Comment (fsodano) [inline index.js:24]: IIFE, fancy.

File: index.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+(function printBoxers() {
+  // debugger;
+  boxerStorage.forEach(e => {
+    const tr = document.createElement('tr');
+    table.appendChild(tr);
+    const boxerData = [
+      [`Name: ${e.firstName} ${e.lastName}`],
+      [`Wins: ${e.wins} `],
+      [` Losses: ${e.losses} `]
```
Comment (fsodano) [inline index.js]: Why did you make these arrays?

File: main.js [hunk 6]
```diff
@@ -3,46 +3,49 @@ saveBoxer.addEventListener('click', createBoxer);
...
+  const newBoxer = {
+    firstName: firstName.value,
+    lastName: lastName.value,
+    nationality: nationality.value,
+    weightClass: weightClass.value,
+    wins: 0,
+    losses: 0
+  };
+  boxers.push(newBoxer);
+  localStorage.setItem('boxers', JSON.stringify(boxers));
```
Comment (fsodano) [inline main.js]: How is this not overriding the existing boxers each time you save it? 🤔

File: main.js [hunk 7]
```diff
@@ -3,46 +3,49 @@ saveBoxer.addEventListener('click', createBoxer);
...
-  localStorage.setItem('boxers', JSON.stringify(newBoxer));
-}
+  const newBoxer = {
+    firstName: firstName.value,
+    lastName: lastName.value,
+    nationality: nationality.value,
+    weightClass: weightClass.value,
+    wins: 0,
+    losses: 0
+  };
```
Comment (fsodano) [inline main.js:30]: You now have entities! Amazeballs. We'll look into these next class.

File: main.js [hunk 8]
```diff
@@ -3,46 +3,49 @@ saveBoxer.addEventListener('click', createBoxer);
...
+  const boxers = JSON.parse(localStorage.getItem('boxers'));
```
Comment (fsodano) [inline main.js]: you really need to get a single function to do this for you mate! You keep repeating this bit of code over and over again.

File: index.html (hunk 1 above)
Comment (stevetait257) [reply index.html]: Are you asking why I renamed it?

File: index.js (hunk 2 above)
Comment (stevetait257) [reply index.js]: Technically, this is a comment.

File: main.js (hunk 8 above)
Comment (stevetait257) [reply main.js]: Yes, I noted that as I have gone on through, I keep needing to do this so will create a function to do this.

File: main.js (hunk 6 above)
Comment (stevetait257) [reply main.js]: It does overwrite them, but with the exact same information. Looking at this now, I think I should have done `localStorage.boxers = JSON.stringify(boxers);`

File: index.js (hunk 5 above)
Comment (stevetait257) [reply index.js]: I cannot answer this. I have undone this unnecessary array and the subsequent spread

**Pull request merged**